compendium/DeclarativeServices/src/manager/states/CCActiveState.cpp

```diff
@@ -52,14 +52,7 @@ std::shared_ptr<ComponentInstance> CCActiveState::Activate(
         }
       });
       std::lock_guard<std::mutex> lock(oneAtATimeMutex);
-      // Make sure the state didn't change while we were waiting
-      auto currentState = mgr.GetState();
-      if (currentState->GetValue() !=
-          service::component::runtime::dto::ComponentState::ACTIVE) {
-        logger->Log(cppmicroservices::logservice::SeverityLevel::LOG_WARNING,
-                    "Activate failed. Component no longer in Active State.");
-        return nullptr;
-      }
+
```

Comment:
Is it possible that while we're waiting for the lock here that the Modified method is running, resulting in Deactivate being called? If so, we might have to keep this check in because the state would have changed while we were waiting.